server/repository/src/db_diesel/goods_received_line_row.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{
+    ChangeLogInsertRow, ChangelogRepository, ChangelogTableName, Delete,
+    PurchaseOrderRowRepository, RepositoryError, RowActionType, StorageConnection, Upsert,
+};
+use chrono::NaiveDate;
+use diesel::prelude::*;
+use diesel_derive_enum::DbEnum;
+use serde::{Deserialize, Serialize};
+
+table! {
+    goods_received_line (id) {
+        id -> Text,
+        goods_received_id -> Text,
+        purchase_order_id -> Text,
+        received_pack_size -> Double,
+        number_of_packs_received -> Double,
+        batch -> Nullable<Text>,
+        weight_per_pack -> Nullable<Double>,
+        expiry_date -> Nullable<Date>,
+        line_number -> BigInt,
+        item_link_id -> Text,
+        item_name -> Text,
+        location_id -> Nullable<Text>,
+        volume_per_pack -> Nullable<Double>,
+        manufacturer_link_id -> Text,
+        // TODO map GoodsReceivedLineStatusMapping in OMS to is_authorized in OG: see issue [8647](https://github.com/msupply-foundation/open-msupply/issues/8068?issue=msupply-foundation%7Copen-msupply%7C8647)
+        status -> crate::db_diesel::goods_received_line_row::GoodsReceivedLineStatusMapping,
+        comment -> Nullable<Text>,
+    }
+}
+
+#[derive(
+    Clone, Insertable, Queryable, Debug, AsChangeset, Serialize, Deserialize, Default, PartialEq,
+)]
+#[diesel(table_name = goods_received_line)]
+#[diesel(treat_none_as_null = true)]
+pub struct GoodsReceivedLineRow {
+    pub id: String,
+    pub goods_received_id: String,
+    pub purchase_order_id: String,
+    pub received_pack_size: f64,
+    pub number_of_packs_received: f64,
+    pub batch: Option<String>,
+    pub weight_per_pack: Option<f64>,
+    pub expiry_date: Option<NaiveDate>,
+    pub line_number: i64,
+    pub item_link_id: String,
+    pub item_name: String,
+    pub location_id: Option<String>,
+    pub volume_per_pack: Option<f64>,
+    pub manufacturer_link_id: String,
+    // TODO map GoodsReceivedLineStatusMapping in OMS to is_authorized in OG: see issue [8647](https://github.com/msupply-foundation/open-msupply/issues/8068?issue=msupply-foundation%7Copen-msupply%7C8647)
```

Comment:
I think this comment should just be in `translations/goods_received_line.rs`
There's no reason for this code here to change...

Reply:
And looks like you've already done the mapping between status and is_authorised anyway, So it's just goods_received that needs to be done, nothing needed for line.